src/cloud-api-adaptor/pkg/userdata/provision.go

```diff
@@ -78,51 +83,57 @@ func (d DefaultRetry) GetRetryDelay() time.Duration {
 type AzureUserDataProvider struct{ DefaultRetry }
 
 func (a AzureUserDataProvider) GetUserData(ctx context.Context) ([]byte, error) {
-	url := azure.AzureUserDataImdsUrl
+	url := AzureUserDataImdsUrl
 	logger.Printf("provider: Azure, userDataUrl: %s\n", url)
-	return azure.GetUserData(ctx, url)
+	return imdsGet(ctx, url, true, []kvPair{{"Metadata", "true"}})
 }
 
 type AWSUserDataProvider struct{ DefaultRetry }
 
 func (a AWSUserDataProvider) GetUserData(ctx context.Context) ([]byte, error) {
-	url := aws.AWSUserDataImdsUrl
+	url := AWSUserDataImdsUrl
 	logger.Printf("provider: AWS, userDataUrl: %s\n", url)
-	return aws.GetUserData(ctx, url)
+	return imdsGet(ctx, url, false, nil)
```

Comment:
shouldn't the bool be true here ?

Reply:
aws doesn't encode the userdata with base64, does it? at least it wasn't in the original code

Reply:
Correct.. I forgot about it.. Do you mind adding a comment above the code ?